hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1299,6 +1302,15 @@ public final HoodieTable initTable(WriteOperationType operationType, Option<Stri
     // Validate table properties
     metaClient.validateTableProperties(config.getProps());
 
+    initTimer(operationType, table);
+
```

Comment:
The Spark set up the timer before starting the writing tasks, while for Flink, it actually starts just before commiting the metadata, is that what you want then?

Reply:
thx for your reply. in fact, the method initTable I didn't do any change, just extract the part for named  init timer method.I actually want to use the init timer method to init the writeTimer used in org.apache.hudi.client.BaseHoodieWriteClient#emitCommitMetrics, when i want to emitCommitMetrics, the writeTimer object have to be not null , or the method emitCommitMetrics will be jumpped over。when i print log ,I find the writeTimer in method [org.apache.hudi.client.BaseHoodieWriteClient#emitCommitMetrics] is null,
 so i put the initTimer method after the [HoodieTable table = createTable(config, hadoopConf);]

Reply:
I find the [Timer.Context writeTimer] in org.apache.hudi.client.BaseHoodieWriteClient seem not to be inited, so i do these changes

Reply:
bulk_writer‘s metric is ok, and we can get metric and report. when flink checkpoint try to commit, it will call the org.apache.hudi.client.BaseHoodieWriteClient#commitStats to report other metrics, however  the writeTimer in BaseHoodieWriteClient is null, when call the method org.apache.hudi.client.BaseHoodieWriteClient#emitCommitMetrics.

Reply:
I kind of think we should init the timer in `StreamWriteOperatorCoordinator#startInstant()`, and we can have a new method:

```java
public void setWriteTimer(String commitType)
```

Reply:
ok，i get you. I try to do that

Reply:
I open another https://github.com/apache/hudi/pull/9637 and fix as above, this maybe close. thx very much